flink-sql-runner-dist/pom.xml

```diff
@@ -0,0 +1,91 @@
+        9999 matches the image's user/group-->
+        <distribution.override.uid>9999</distribution.override.uid>
+        <distribution.override.gid>9999</distribution.override.gid>
+    </properties>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>com.googlecode.maven-download-plugin</groupId>
+                <artifactId>download-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <phase>process-resources</phase>
+                        <goals>
+                            <goal>wget</goal>
+                        </goals>
+                        <configuration>
+                            <url>${flink.download.url}</url>
+                            <sha512>${flink.download.sha512}</sha512>
+                            <unpack>true</unpack>
+                            <alwaysVerifyChecksum>true</alwaysVerifyChecksum>
+                            <outputDirectory>${project.build.directory}/flink-distribution</outputDirectory>
+                            <!--observed GitHub actions hitting the read timeout even at 20seconds-->
+                            <readTimeOut>60000</readTimeOut>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <artifactId>maven-assembly-plugin</artifactId>
+                <configuration>
+                    <descriptors>
+                        <descriptor>src/assembly/binary-assembly.xml</descriptor>
+                    </descriptors>
+                    <overrideGid>${distribution.override.gid}</overrideGid>
+                    <overrideUid>${distribution.override.uid}</overrideUid>
+                </configuration>
+                <executions>
+                    <execution>
+                        <id>make-assembly</id>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>single</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+    <dependencies>
+        <!--included in /opt/streamshub/-->
+        <dependency>
+            <groupId>com.github.streamshub</groupId>
+            <artifactId>flink-sql-runner</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <!--included in /opt/streamshub/lib/-->
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-sql-avro-confluent-registry</artifactId>
+            <version>${flink.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-sql-connector-kafka</artifactId>
+            <version>${flink.kafka.connector.version}</version>
+        </dependency>
```

Comment:
I'm sure we've talked about this somewhere else but: What is the conceptual difference between these dependencies and the ones we shade into the runner?

Reply:
You mean the difference between flink connectors and something like the fabric8 kubernetes client?

The difference is flink is able to load arbitrary connectors and formats off the classpath. So we can create an SQL runner jar that contains no specific connectors/formats. The job submits SQL to Flink, which can make use of whatever connectors or formats are available in the flink environment.

For example, to support creating a Pulsar sourced table, we don't have to rebuild the flink job jar if the pulsar connector is part of the environment.

However the other integrations, like with the k8s API through fabric8 are compile dependencies of the job. Conceptually I think those classes are part of the job's responsibilities, where the connectors/formats loaded by Flink and accessed via SQL are environmental, they just usually happen to be typically loaded from the job jar.

Reply:
Does that make any sense? The sql runner job is not directly dependant on any particular connector. It supplies an SQL string that might need a particular connector available to function.

Reply:
Still not 100% sold on shading everything vs not, but I think I'll merge and we can revisit. If we shaded everything users would still be able to put further connectors/formats on the classpath if they require them.

Reply:
mm, in future it looks like they might support connectors/formats being installed into the plugins dir which would be cool, then they'd get an isolated classloader

https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/filesystems/plugins/#plugins

Reply:
Yes that finally makes sense to me. 

So the rule would be if its provided scope (at least conceptually) in in `flink-sql-runner/pom.xml` it goes in here.